pkg/csi/cinder/identityserver.go

```diff
@@ -47,11 +46,14 @@ func (ids *identityServer) GetPluginInfo(ctx context.Context, req *csi.GetPlugin
 }
 
 func (ids *identityServer) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
-	_, err := openstack.CreateOpenStackProvider()
-	if err != nil {
-		klog.V(3).Infof("Failed to CreateOpenStackProvider: %v", err)
-		return nil, status.Error(codes.FailedPrecondition, "Failed to communicate with openstack")
-	}
+
+	// Removing openstack check here, as it it done in cmd
```

Comment:
should we retain this code here as well? i think probe might be called at any time / multiple times to check readiness , but in cmd , we only check the health at the driver start?

Reply:
@ramineni the problem here is` CreateOpenStackProvider()` is not an interface method, so i cannot fake it while doing the sanity testing, it is causing probe failure in sanity tests. 
IMO in probe, creating the whole openstack provider is not necessary, we can just check the Cinder root API or any GET API. I will propose a seperate PR for the Probe Implementation.

Reply:
@adisky ya , I agree , if probe will get called multiple times, no need to create provider everytime to check readiness I suppose. please log seperate issue to fix the same.
@jichenjc Thoughts?

Reply:
+1 to separate issue ~

Reply:
@ramineni @jichenjc Issue Logged 
https://github.com/kubernetes/cloud-provider-openstack/issues/637

Reply:
Uncommented the current code